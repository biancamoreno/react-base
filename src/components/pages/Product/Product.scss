.product {
    width: 80%;
    margin: 0 auto;
}

.product_container {
    display: flex;
}

.product_display {
    display: flex;
    padding: 32px;
}

.product_big-image-container {
    display: flex;
    flex: 0 1 60%;
    justify-content: center;
    background-color: white;
}

.product_detail {
    display: flex;
    flex: 0 1 40%;
    padding: 0 32px;
    flex-direction: column;
    background-color: white;
}

.product_quantity {
    padding: 32px 0 16px 0;
    font-size: 14px;
}

.product_seller-name {
    font-size: 24px;
    padding-bottom: 32px;
}

.product_price {
    font-size: 46px;
    font-weight: bold;
    padding-bottom: 32px;   
}

.product_description-container {
    padding: 32px;
    background-color: white;
}

.product_description-title {
    font-size: 28px;
}

.product_description {
    font-size: 16px;
    padding: 32px 0;
}

.product_pay-button {
    height: 32px;
    max-width: 200px;
    color: white;
    background-color: #3483fa;
    cursor: pointer;
    text-align: center;
    border-radius: 4px;
    outline: none;
    border: none;
}