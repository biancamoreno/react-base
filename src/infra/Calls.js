import axios from 'axios';

const baseUrl = 'https://api.mercadolibre.com/sites/MLA/search?q=';
const idSearcUrl = 'https://api.mercadolibre.com/items/';
const limit = 10;

export const getProductsByName = (productName) => {
    return axios.get(`${baseUrl}${productName}&limit=${limit}`);
}

export const getInitialSearch = () => {
    return axios.get(`${baseUrl}skate&limit=${limit}`);
}

export const getProductById = (id) => {
    return axios.get(`${idSearcUrl}${id}`);
}

export const getDetailProductById = (id) => {
    return axios.get(`${idSearcUrl}:${id}/description`);
}
