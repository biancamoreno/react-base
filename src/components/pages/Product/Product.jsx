import React, { Component } from "react";
import "../Product/Product.scss";
import { getProductById } from "../../../infra/Calls";

export class ProductPage extends Component {

  constructor(props) {
    super(props);

    this.state = {
      product: null
    }
  }

  handleSearchById(id) {
    getProductById(id)
        .then(res => {
            const product = res.data;
            this.setState({ product });
        })
        .catch(error => console.log(error))
  }

  componentDidMount () {
    const { id } = this.props.match.params;
    this.handleSearchById(id);
  }

  render() {
    const { id } = this.props.match.params;

    return (
      <>
        <p>
          <strong>Match Props: {id} </strong>
        </p>
      </>
    );
  }
};

export default ProductPage;
