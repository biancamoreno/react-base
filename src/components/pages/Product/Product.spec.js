import React from 'react';
import toJson from 'enzyme-to-json'
import { shallow } from 'enzyme';
import Product from './Product';
import { getProductById, getDetailProductById } from "../../../infra/Calls";


describe('Product test', () => {

    const defaultProps = {
        match: {
            params: { id: "MLA820795723" }
        }
    }

    const defaultState = {
        product: {
            available_quantity: 1,
            condition: "new",
            currency_id: "ARS",
            id: "MLA820795723",
            initial_quantity: 16,
            pictures: [{
                0: { secure_url: "https://mla-s2-p.mlstatic.com/951232-MLA31003080341_062019-O.jpg" }
            }],
            domain_id: "MLA-CELLPHONES",
            price: 129999,
        },
        descriptions: {
            plain_text: "Una mente brillante.â†µEl iPhone 8 es ..."
        }
    }

    it('should render correctly with state', () => {
        const wrapper = shallow(<Product {...defaultState} {...defaultProps} />)
        expect(toJson(wrapper)).toMatchSnapshot()
    });

});