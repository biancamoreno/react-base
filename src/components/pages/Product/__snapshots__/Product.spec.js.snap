// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product test should render correctly with state 1`] = `
<Fragment>
  <div
    className="product"
  />
</Fragment>
`;
